solana/programs/example-native-token-transfers/src/instructions/redeem.rs

```diff
@@ -134,7 +134,7 @@ pub fn redeem(ctx: Context<Redeem>, _args: RedeemArgs) -> Result<()> {
         RateLimitResult::Delayed(release_timestamp) => release_timestamp,
     };
 
-    accs.inbox_item.release_status = ReleaseStatus::ReleaseAfter(release_timestamp);
+    accs.inbox_item.release_after(release_timestamp)?;
```

Comment:
Could this be moved to L114 (above the rate limit check)? Since `release_after` has the possibility to return early, we might as well avoid doing the rate limiting calculations

Reply:
agree it would be more efficient to perform the check first, then do the calculation, then set the value. I was thinking this way it's a little more atomic (since the setting happens in the same function as the assertion)

Reply:
If not done atomically via your `release_after` method, might be nice to have it as a constraint in the account context, something like:

```rust
        constraint = inbox_item.release_status == ReleaseStatus::NotApproved @ NTTError::TransferCannotBeRedeemed
```

Just so it's clear from reading the instruction's account context that it cannot be redeemed if it isn't approved yet. This works because `NotApproved` is serialized as 0 and this byte is 0 when this account is initialized.

Reply:
sounds good, I'll add that in a follow up PR